persistence/src/main/java/de/cyface/persistence/PersistenceLayer.java

```diff
@@ -745,7 +745,13 @@ private Track collectNextSubTrack(@NonNull final Cursor geoLocationCursor, @NonN
         while (location.getTimestamp() <= pauseEventTime) {
 
             track.add(location);
-            geoLocationCursor.moveToNext();
+
+            // Stop if this is the last GeoLocation
+            if (!geoLocationCursor.moveToNext()) {
+                break;
+            }
+
+            // Load next GeoLocation to check it's timestamp in next iteration
             location = loadGeoLocation(geoLocationCursor);
```

Comment:
You could also just let loadGeoLocation return null if there is no further location and finish the loop as soon as location becomes null. This would prevent the above mentioned problem.